### ISSUE TITLE
regex url matching doesn't work

### ISSUE THREAD

**Issue by brycedrennan** [h2non/pook#45]
Test case
```python
import pook

def test_pook_regex():
    pook.on()
    # from the example docs: http://pook.readthedocs.io/en/latest/examples.html?highlight=regex#regular-expression-matching
    pook.get(pook.regex('h[t]{2}pbin.*'))
```

Result
```
    @url.setter
    def url(self, url):
>       if not protoregex.match(url):
E       TypeError: expected string or bytes-like object

../.direnv/python-3.6.2/lib/python3.6/site-packages/pook/request.py:90: TypeError
```

**Comment by h2non** [comment]
Thanks for reporting. This should be fixed in `v0.2.4`. Please, upgrade it running:
```bash
pip install --upgrade pook
```

**Comment by brycedrennan** [comment]
Thanks!
